malformed manifest causes error; but helm doesn't point to the issue

When running `helm install` on a chart with a malformed YAML manifest, I get:

```
><> helm install workflow-beta2
[ERROR] Failed to load chart: error converting YAML to JSON: yaml: line 36: did not find expected key
```

This issue did not help with identifying *which* manifest was to blame.

--- a/manifest/manifest.go
+++ b/manifest/manifest.go
@@ -66,13 +66,13 @@ func Parse(filename string) ([]*Manifest, error) {
 
 	docs, err := codec.YAML.Decode(d).All()
 	if err != nil {
-		return ms, fmt.Errorf("%s %s", filename, err)
+		return ms, fmt.Errorf("%s: %s", filename, err)
 	}
 
 	for _, doc := range docs {
 		ref, err := doc.Meta()
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("%s: %s", filename, err)
 		}
 
 		m := &Manifest{Version: ref.APIVersion, Kind: ref.Kind, Name: ref.Name, VersionedObject: doc, Source: filename}

--- a/manifest/manifest_test.go
+++ b/manifest/manifest_test.go
@@ -1,6 +1,7 @@
 package manifest
 
 import (
+	"strings"
 	"testing"
 )
 
@@ -55,5 +56,15 @@ func TestParseDir(t *testing.T) {
 		if man.Name == "" {
 			t.Error("Expected name")
 		}
+	}
+
+	// now test parsing bad files in a chart!
+	testchart = "../testdata/charts/malformed"
+	manifests, err = ParseDir(testchart)
+	if err == nil {
+		t.Errorf("Failed to raise an error when parsing dir %s", testchart)
+	}
+	if !strings.Contains(err.Error(), "malformed.yaml") {
+		t.Errorf("Failed to identify which manifest failed to be parsed. Got %s", err)
 	}
 }

--- a/testdata/charts/malformed/Chart.yaml
+++ b/testdata/charts/malformed/Chart.yaml
@@ -1,0 +1,11 @@
+name: malformed
+home: http://github.com/helm/helm
+source:
+  - https://example.com/helm
+version: 0.0.1
+description: All the things, all malformed, none working
+maintainers:
+  - Helm <helm@deis.com>
+details:
+  This package provides poorly formatted manifests.
+  It can be used to test how helm parses a chart.

--- a/testdata/charts/malformed/manifests/malformed.yaml
+++ b/testdata/charts/malformed/manifests/malformed.yaml
@@ -1,0 +1,6 @@
+apiVersion: v1
+kind: Pod
+spec:
+  containers:
+  - name: "look-ma-no-end-quote!
+    image: alpine:3.2

